models/plan.js

```diff
@@ -4,86 +4,116 @@
 const pool = require("../services/db/mysqlPool").pool;
 
 // save a plan with its selected courses. remove the plan if an error occurs
-async function savePlan(userId, planName, courses) {
+async function createPlan(userId, planName, courses) {
 
   try {
 
-    // save the basic plan information
-    const planId = await insertPlan(userId, planName);
+    // construct the first SQL query
+    let sql = "BEGIN; " +
+      "INSERT INTO Plan (studentId, planName, status) VALUES (?, ?, 2); " +
+      "SELECT LAST_INSERT_ID();";
```

Comment:
I wonder why you did this, since the result of the `INSERT INTO` query is the `insertId` of the new object.

Reply:
It is a transaction. While it looks like a single insert query, it is not. I will not get the insert value back until EVERYTHING is ran, meaning I need someway to store the ID mid transaction. This is the way to do that. This allows me to insert the base plan and the selected courses atomically, meaning that if the database crashes halfway through the process we don't end-up with a plan with no courses.